src/mscorlib/src/System/String.cs

```diff
@@ -2174,43 +2175,48 @@ public static int Compare(String strA, String strB, StringComparison comparisonT
         // if this is equal to value, or a value greater than 0 if this is greater than value.
         //
         [Pure]
-        public int CompareTo(Object value) {
-            if (value == null) {
+        public int CompareTo(Object value)
+        {
+            if (value == null)
+            {
                 return 1;
             }
 
-            if (!(value is String)) {
+            string other = value as string;
+
+            if (other == null)
+            {
                 throw new ArgumentException(Environment.GetResourceString("Arg_MustBeString"));
             }
-
-            return String.Compare(this,(String)value, StringComparison.CurrentCulture);
+            
```

Comment:
@jasonwilliams200OK IMO I'm not sure that the whitespace changes matter all that much, as long as they're spaces and they're either aligned with the braces or the beginning of the line. GitHub has a trick where you can append ?w=1 to the end of the URL to eliminate the whitespace changes in a diff, e.g.:

https://github.com/dotnet/coreclr/pull/6603/files?w=1